borg/remote.py

```diff
@@ -316,22 +316,18 @@ def handle_error(error, res):
             if w:
                 while not self.to_send and (calls or self.preload_ids) and len(waiting_for) < 100:
                     if calls:
-                        if is_preloaded:
-                            if calls[0] in self.cache:
-                                waiting_for.append(fetch_from_cache(calls.pop(0)))
```

Comment:
I have doubts that removing this is beneficial:

The difference between the if and else blocks seems to be that if is_preloaded is True, we know that all the content data chunks have been queued for preloading (borg extract does this). Thus we never do request them. If we do not already have them in the cache, it just means that the preload request was not processed / answered YET (but will be later).

In the "else" branch (is_preloaded is False), we actively request anything we have not in the cache yet because we know that no preload was queued for this chunk, so we need to do the request here.